homeassistant/components/modbus/modbus.py

```diff
@@ -310,6 +310,13 @@ def _log_error(self, text: str, error_state: bool = True) -> None:
             _LOGGER.error(log_text)
             self._in_error = error_state
 
+    async def async_pymodbus_connect(self) -> None:
+        """Connect to device, async."""
+        async with self._lock:
+            if not await self.hass.async_add_executor_job(self._pymodbus_connect):
```

Comment:
Hmm, `async_add_executor` tasks are also blocking startup. I need to think about if this is a mistake in our API design.